codec/helper.go

```diff
@@ -437,7 +437,7 @@ var immutableKindsSet = [32]bool{
 	// reflect.Ptr
 	// reflect.Slice
 	reflect.String: true,
-	// reflect.Struct
+	reflect.Struct: true,
```

Comment:
Do we have any handle on the performance impact of this? Could it lead to a lot of garbage if we decode elements into a large struct 1-by-1? (I haven't dug into the guts yet, so not sure if that's what it does)

Reply:
It definitely has a performance impact, but I haven't measured it. I don't feel comfortable with the unsafe code path, since it requires accessing the internal parts of pointers: https://github.com/hashicorp/go-msgpack/blob/v1.1.5/codec/helper_unsafe.go#L342-L346

Reply:
I think it would be valuable to downstream consumers to know a ballpark level of impact so that they can take action if needed, but in general I'm fine with taking a performance hit for safety and maintainability.